src/components/Header.tsx

```diff
@@ -2,80 +2,106 @@ import React, { useState } from 'react';
 import { Navigation } from 'hds-react';
 import { useHistory, useLocation } from 'react-router-dom';
 import { useClient } from '../client/hooks';
-import config from '../config';
 import styles from './styles.module.css';
+import { getClientConfig } from '../client';
+import config from '../config';
 
 type Page = 'frontpage' | 'apiAccessTokens' | 'userTokens' | 'profile';
 
 const Header = (): React.ReactElement => {
+  const currentConfig = getClientConfig();
+  const pathPrefix = currentConfig.path;
+  const renderApiTokenLink = currentConfig.hasApiTokenSupport;
   const client = useClient();
   const authenticated = client.isAuthenticated();
   const initialized = client.isInitialized();
   const user = client.getUser();
   const history = useHistory();
   const location = useLocation();
+  const path = location.pathname.replace(pathPrefix, '');
   const currentPageFromPath: Page =
-    location.pathname && location.pathname.length > 1
-      ? (location.pathname.substr(1) as Page)
-      : 'frontpage';
+    path && path.length > 1 ? (path.substr(1) as Page) : 'frontpage';
   const [active, setActive] = useState<Page>(currentPageFromPath);
 
   const title = 'Helsinki Profiili Example';
   const userName = user ? `${user.given_name} ${user.family_name}` : '';
 
+  const frontPageLink = (
+    <Navigation.Item
+      active={active === 'frontpage'}
+      label="Etusivu"
+      key="frontpage"
+      tabIndex={0}
+      onClick={(): void => {
+        setActive('frontpage');
+        history.push(pathPrefix);
+      }}
+      data-test-id="header-link-frontpage"
+    />
+  );
+
+  const accessTokenLink = (
+    <Navigation.Item
+      active={active === 'apiAccessTokens'}
+      as="button"
+      label="Hae API access token"
+      key="apiAccessTokens"
+      type="button"
+      onClick={(): void => {
+        setActive('apiAccessTokens');
+        history.push(`${pathPrefix}/apiAccessTokens`);
+      }}
+      data-test-id="header-link-apiAccessTokens"
+    />
+  );
+
+  const userTokenLink = (
+    <Navigation.Item
+      active={active === 'userTokens'}
+      as="button"
+      label="Tokenit"
+      key="userTokens"
+      type="button"
+      onClick={(): void => {
+        setActive('userTokens');
+        history.push(`${pathPrefix}/userTokens`);
+      }}
+    />
+  );
+
+  const profileLink = (
+    <Navigation.Item
+      active={active === 'profile'}
+      as="button"
+      label="Profiili"
+      key="profile"
+      type="button"
+      onClick={(): void => {
+        setActive('profile');
+        history.push(`${pathPrefix}/profile`);
+      }}
+      data-test-id="header-link-profile"
+    />
+  );
+
+  // <Navigation.Row> cannot not handle null/undefined as children
+  // cannot use if..else in <Navigation.Row>, so must define children as array
```

Comment:
Some repeated words in this comment. Also the wording could be better, I'm not really sure what this is trying to communicate.

Reply:
Added some more words :)